featuretools/primitives/standard/transform_primitive.py

```diff
@@ -863,3 +864,45 @@ def email_address_to_domain(emails):
                 emails_df['domain'] = emails_df['email'].str.strip().str.split('@', expand=True)[1]
             return emails_df.domain.values
         return email_address_to_domain
+
+
+class NumericLag(TransformPrimitive):
+    """Shifts an array of values by a specified number of periods.
+
+    Args:
+        periods (int): The number of periods by which to shift the input.
+            Default is 1. Periods correspond to rows.
+
+        fill_value (int, float, optional): The value to use to fill in
+            the gaps left after shifting the input. Default is None.
+
+    Examples:
+        >>> lag = NumericLag()
+        >>> lag(pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D')), [1, 2, 3, 4, 5]).tolist()
+        [nan, 1.0, 2.0, 3.0, 4.0]
+
+        You can specify the number of periods to shift the values
+
+        >>> lag_periods = NumericLag(periods=3)
+        >>> lag_periods(pd.Series(pd.date_range(start="2020-01-01", periods=5, freq='D')), [1, 2, 3, 4, 5]).tolist()
+        [nan, nan, nan, 1.0, 2.0]
+
+        You can specify the fill value to use
+
+        >>> lag_fill_value = NumericLag(fill_value=100)
+        >>> lag_fill_value(pd.Series(pd.date_range(start="2020-01-01", periods=4, freq='D')), [1, 2, 3, 4]).tolist()
+        [100, 1, 2, 3]
+    """
+    name = "numeric_lag"
+    input_types = [ColumnSchema(semantic_tags={'time_index'}), ColumnSchema(semantic_tags={'numeric'})]
```

Comment:
Is there a reason for requiring a time index as input to this primitive?

Reply:
Having a lagging feature only really makes sense if the data has a temporal ordering to it.

Otherwise a `NumericLag` feature with `periods=k` is essentially just choosing to include other row values from the dataset without any rhyme or reason other than the fact that they just happened to be the row that is `k` periods away.

Reply:
Agreed, but we aren't actually doing anything with the ordering inside the primitive. So, I guess my question maybe is this - should we require the user to have a time index set and have FT pass it to the primitive unused, or should we document the behavior of this primitive and expect the user to use it properly only when they have correctly ordered data?

I'm not really advocating for one or the other at this point, just bringing it up for discussion.

Reply:
We do the same thing with the rolling primitives, though in the future we will use pandas' offset aliases, which will require those datetime values. So that might have been a good enough reason to require the time index. 

We could do the same thing here, actually, and at some point allow the lag primitive to be used with offset aliases.